plugins/inputs/snmp/snmp.go

```diff
@@ -535,8 +538,8 @@ func (t Table) Build(gs snmpConnection, walk bool) (*RTable, error) {
 		Time: time.Now(), //TODO record time at start
 		Rows: make([]RTableRow, 0, len(rows)),
 	}
-	for _, r := range rows {
-		if len(r.Tags) < tagCount {
+	for i, r := range rows {
+		if len(r.Tags)+emptyTags[i] < tagCount {
 			// don't add rows which are missing tags, as without tags you can't filter
 			continue
 		}
```

Comment:
Okay, so I've implemented this and I would doubt that we want it that way. Anyway, I don't have a very strong opinion as long as I get my measurements back :-)

https://github.com/hahnjo/telegraf/commit/9699fcac21706619ddd33124626f01f17fd43ce0 (look out for the test changes and especially `rtr4` which is added although the tag doesn't even exist in SNMP)